dms.py

```diff
@@ -115,6 +118,90 @@ def render_page(l):
 img.redflag {
     background-color: orange;
 }
+
+#search {
+    width: 800px;
+    margin-left: auto;
+    margin-right: auto;
+    display: grid;
+    grid-template-columns: auto 200px;
+    grid-template-rows: auto;
+    grid-template-areas: "input button";
+}
+#search > .input {
+    grid-area: input;
+    margin-right: 5px;
+    padding-left: 11px;
+}
+#search > .button {
+    grid-area: button;
+}
+
+#edit {
+    width: 800px;
+    margin-left: auto;
+    margin-right: auto;
+    display: grid;
+    grid-template-columns: 220px 580px;
+    grid-template-rows: auto;
+    grid-template-areas:
+    "description description"
+    "downloads preview";
+}
+#edit > .description {
+    grid-area: description;
+}
+#edit > .list_items{
+    grid-area: downloads;
+    margin-left:auto;
+    padding: 0px;
+}
+#edit > .preview{
+    grid-area: preview;
+    margin-left:auto;
+    margin-right:auto;
+}
+
+.description {
+    display: grid;
+    grid-template-columns: auto 200px;
+    grid-template-rows: auto;
+    grid-template-areas:
+    "input input"
+    ". button";
+}
+.description > .input {
+    grid-area: input;
+    height: 100px;
+    padding-left: 10px;
+    padding-top: 8px;
+    padding-right: 6px;
+}
+.description > .button{
+    grid-area: button;
+    margin-top: 5px;
+}
+
+.button {
+    background-color: #4CAF50; /* Green */
+    color: white;
+    transition-duration: 0.4s;
+    border: none;
+    padding: 12px 28px;
+    text-decoration: none;
+}
+.button:hover {
+    background-color: #dcefdc;
+    color: #4CAF50;
+}
+
+.list_items {
+    list-style: none;
+}
+.list_items li {
+    margin-bottom: 30px;
```

Comment:
Please do not use pixel-sized margins in CSS. Please use a different unit like em or rem.

Reply:
I don't know how to work with em or rem. In my experience it is hard to create a layout using em and rem.
Why should every thing scaled relative to the font? https://www.w3schools.com/css/css_units.asp

Reply:
Because the 4 pixels are much more on a 75 dpi display than on a high res iphone or 4K monitor.

Reply:
https://www.w3schools.com/css/css_units.asp

Pixels (px) are relative to the viewing device. For low-dpi devices, 1px is one device pixel (dot) of the display. For printers and high resolution screens 1px implies multiple device pixels.

Reply:
This is not correct for all cases. Please refer to https://hacks.mozilla.org/2013/09/css-length-explained/ and https://medium.com/@julienetienne/pixels-are-dead-faa87cd8c8b9

Reply:
The first article is 7 years old (for css time heavily outdated), and the second one is saying : 
> The device pixel ratio depicts the size of the optical reference unit and that depends only on the device or the desktop zoom level set by the user.

Anyway I don't know how to use em and rem.